### PR TITLE
Some refactoring and optimization

### PR DESCRIPTION
**Pull request by adhi1419**

### CODE REVIEW

File: msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/Arrays.java [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import java.util.ArrayList;
+
+import it.unimi.dsi.fastutil.ints.IntComparator;
```
Comment (tomas-pluskal) [inline msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/Arrays.java]: Is this import going to work if you removed the dsiutils library?

File: msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/Arrays.java [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+   * standard mergesort, as its running time is
+   * <i>O</i>(<var>n</var>&nbsp;(log&nbsp;<var>n</var>)<sup>2</sup>), but it does not allocate
+   * additional memory; as a result, it can be used as a generic sorting algorithm.
+   * 
+   * @param from the index of the first element (inclusive) to be sorted.
+   * @param to the index of the last element (exclusive) to be sorted.
+   * @param c the comparator to determine the order of the generic data (arguments are positions).
+   * @param swapper an object that knows how to swap the elements at any two positions.
+   */
+  public static void mergeSort(final int from, final int to, final IntComparator c,
```
Comment (tomas-pluskal) [inline msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/Arrays.java]: Do you need all these utility functions like mergeSort?

File: msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/BooleanArrays.java [hunk 3]
```diff
@@ -0,0 +1,844 @@
...
+ *
+ * <P>
+ * Note that {@link it.unimi.dsi.fastutil.io.BinIO} and {@link it.unimi.dsi.fastutil.io.TextIO}
+ * contain several methods make it possible to load and save arrays of primitive types as sequences
+ * of elements in {@link java.io.DataInput} format (i.e., not as objects) or as sequences of lines
+ * of text.
+ *
+ * @see java.util.Arrays
+ */
+public class BooleanArrays {
```
Comment (tomas-pluskal) [inline msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/BooleanArrays.java]: I think you don't need this class, or do you?

File: msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/Arrays.java (hunk 1 above)
Comment (adhi1419) [inline msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/Arrays.java]: Oops, will fix it.

File: msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/BooleanArrays.java (hunk 3 above)
Comment (adhi1419) [inline msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/BooleanArrays.java]: https://github.com/adhi1419/msdk/blob/449dadad7321c17eeebb4f9d8efcc870bbf7d116/msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/ByteBufferInputStream.java#L135

**Comment by tomas-pluskal** [comment]
By convention package names in Java are always lower case, so please rename the ByteBuffer package to bytebuffer (or maybe bytebufferinputstream, makes more sense)

File: msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/BooleanArrays.java (hunk 3 above)
Comment (TDI-tenderholt) [inline msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/ByteBuffer/BooleanArrays.java]: You could use https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#fill(boolean[],%20boolean) instead? Then you don't need this class.

**Pull request merged**